pkg/graveler/staging/db_manager.go

```diff
@@ -113,3 +137,81 @@ func (p *DBManager) txOpts(opts ...db.TxOpt) []db.TxOpt {
 	}
 	return append(o, opts...)
 }
+
+// runWorker writing staging token data to fd
+func runWorker(ctx context.Context, d *pgxpool.Pool, jobChan <-chan string) error {
+	record := &graveler.ValueRecord{}
+	for t := range jobChan {
```

Comment:
The data skew here worries me. It's not uncommon to have very few branches of lakeFS, with only one the branches containing hundred of thousands of objects. Paralle5lizing per branch won't really help in these situation, we would have to parallelize within a staging token.

This comment is an optimization one for sure, we can flag it as premature and open an issue. I believe we should also recommend the users to commit their changes before the migration to speed it up dramatically.

Reply:
I think it's a very good point on the commit.
I was using the 'natural' division of labor for the initial concurrency and was deliberating about this exact point. I agree we should open it as an improvement, we just need to see how we ensure we assign the right amount of workers for each workload.